server/internal/config/config.go

```diff
@@ -216,3 +272,343 @@ func getNetwork() (common.Network, error) {
 		return common.Network{}, fmt.Errorf("unknown network %s", viper.GetString(Network))
 	}
 }
+
+func DetermineLocktimeType(locktime int64) common.Locktime {
+	if locktime >= 512 {
+		return common.Locktime{Type: common.LocktimeTypeSecond, Value: uint32(locktime)}
+	}
+
+	return common.Locktime{Type: common.LocktimeTypeBlock, Value: uint32(locktime)}
+}
+
+func (c *Config) Validate() error {
+	if !supportedEventDbs.supports(c.EventDbType) {
+		return fmt.Errorf("event db type not supported, please select one of: %s", supportedEventDbs)
+	}
+	if !supportedDbs.supports(c.DbType) {
+		return fmt.Errorf("db type not supported, please select one of: %s", supportedDbs)
+	}
+	if !supportedSchedulers.supports(c.SchedulerType) {
+		return fmt.Errorf("scheduler type not supported, please select one of: %s", supportedSchedulers)
+	}
+	if !supportedTxBuilders.supports(c.TxBuilderType) {
+		return fmt.Errorf("tx builder type not supported, please select one of: %s", supportedTxBuilders)
+	}
+	if len(c.UnlockerType) > 0 && !supportedUnlockers.supports(c.UnlockerType) {
+		return fmt.Errorf("unlocker type not supported, please select one of: %s", supportedUnlockers)
+	}
+	if c.RoundInterval < 2 {
+		return fmt.Errorf("invalid round interval, must be at least 2 seconds")
+	}
+	if !supportedNetworks.supports(c.Network.Name) {
+		return fmt.Errorf("invalid network, must be one of: %s", supportedNetworks)
+	}
+	if c.RoundLifetime.Type == common.LocktimeTypeBlock {
+		if c.SchedulerType != "block" {
+			return fmt.Errorf("scheduler type must be block if round lifetime is expressed in blocks")
+		}
+	} else { // seconds
+		if c.SchedulerType != "gocron" {
+			return fmt.Errorf("scheduler type must be gocron if round lifetime is expressed in seconds")
+		}
+
+		// round life time must be a multiple of 512 if expressed in seconds
+		if c.RoundLifetime.Value%minAllowedSequence != 0 {
+			c.RoundLifetime.Value -= c.RoundLifetime.Value % minAllowedSequence
+			log.Infof(
+				"round lifetime must be a multiple of %d, rounded to %d",
+				minAllowedSequence, c.RoundLifetime,
+			)
+		}
+	}
+
+	if c.UnilateralExitDelay.Type == common.LocktimeTypeBlock {
```

Comment:
Sorry guys, but I must insist on this.

If you look at code, and you pass to config a UnilateralExitDelay of 666, from `func DetermineLocktimeType()` the `c.UnilateralExitDelay.Type` will be `common.LocktimeTypeBlock` so `Validate()` will always throw here.

This means we will never be able to pass a value >= 512 to UnilateralExitDelay.

Am I wrong?

Reply:
no if you pass 666, it will be read as `common.LocktimeTypeSecond`.

That's the expected behaviour. We don't have a flag to signal what type of lockime we want to use, so we are using the value itself:
* if `< 512` = we admit it's a "block number" locktime
* if `> 512` = we admit it's a "second" locktime